packages/private/edtr-io/src/plugins/types/text-exercise-group.tsx

```diff
@@ -39,6 +39,7 @@ export const textExerciseGroupTypeState = entityType(
   {
     ...entity,
     content: editorContent(),
+    cohesive: boolean(false),
```

Comment:
We should probably add a migration for that (even though this is probably handled correctly thanks to `undefined` being falsy).

Reply:
Ah, yes, `deserialize.ts` is already failing because of this. You can handle that there.

Reply:
How can I test this?

Reply:
Ah, I see...

Reply:
> How can I test this?

Generally: by checking what happens when you edit / render the state of existing Edtr.io documents. In this specific case there is probably no need for that since TypeScript already forbids you to pass an invalid state (types ftw). If this was more complicated, one approach might be to test that `deserialize` outputs the correct new state for existing Edtr.io documents. Alternatively: add a db migration that updates all existing Edtr.io documents to the new states.

Reply:
Okay, fixed... :smile:

Reply:
Actually the line `const isCohesive = cohesive.value ?? false` is the migration. Is this okay or shall I put this somewhere else?

Reply:
That line itself isn't doing anything because Edtr.io already makes sure that you get the correct type. So as a plugin developer, you can always assume that the types are as they are defined in your state (and what TypeScript tells you they are).

The migration itself should happen somewhere else. There are a couple of approaches for this:
- On database level > database migration
- On server-side > in `deserialize.ts`
- On client-side > by using the `migration` state helper Edtr.io exposes

Reply:
Actually I got `cohesive.value === undefined` in my plugin. Thus the line `const isCohesive = cohesive.value ?? false`. Is this intended?

Reply:
That shouldn't happen, so that sounds like a bug. So I'm very interested in a reproducable example here :)